src/components/common/filtering/useProject.tsx

```diff
@@ -0,0 +1,145 @@
+import { useCallback, useEffect, useState } from "react";
+import agent from "../../../utils/agent";
+import { Project, ProjectLite, ProjectParams } from "./project";
+
+export default function useProjects() {
+  const [projects, setProjects] = useState<ProjectLite[]>([]);
+  const [projectsLoaded, setProjectsLoaded] = useState(false);
+  const [projectsLoading, setProjectsLoading] = useState(false);
+  const [filteredProjects, setFilteredProjects] = useState<ProjectLite[]>([]);
+  const [positions, setPositions] = useState<string[]>([]);
+  const [positionsLoaded, setPositionsLoaded] = useState(false);
+  const [commitments, setCommitments] = useState<string[]>([]);
+  const [commitmentsLoaded, setCommitmentsLoaded] = useState(false);
+
+  const SetProjectParams = (value: ProjectParams) => {
+    setFilteredProjects(FilterProjects(projects, value));
+  };
+
+  /**
+   * Request the projects from the API
+   */
+  const fetchProjects = useCallback(async () => {
+    try {
+      const result: Project[] = await agent.Projects.list();
+      if (result.length > 0) {
+        const list = result.map((item: Project) => ({
+          id: item.id,
+          slug: item.slug,
+          catchPhrase: item.catchPhrase,
+          title: item.title,
+          description: item.description,
+          commitmentLevel: item.commitmentLevel,
+          openPositions: item.openPositions,
+        }));
+        setProjects(list);
+        setFilteredProjects(list);
```

Comment:
why do we "setFiltertedProjects" here? "setProjects" already does the same thing!

Reply:
the project array contain the complete project entities, and filtered project is an array that will contain the filtered projects, if we dont keep a full list of project, we will have to call the api each time a filter change, so I kept a copy of minified project list to avoid calling the API

Reply:
the first fetch is always with an empty filters criteria so I copy the projects directly to the Filtered projects array